codectrl-server/src/lib.rs

```diff
@@ -500,6 +507,49 @@ impl Authentication for Service {
     ) -> Result<Response<Token>, Status> {
         todo!()
     }
+
+    async fn github_login(
+        &self,
+        _request: Request<GhLogin>,
+    ) -> Result<Response<LoginUrl>, Status> {
+        Ok(Response::new(LoginUrl {
+            url: generate_github_login_url(),
+        }))
+    }
+}
+
+fn generate_github_login_url() -> String {
+    let github_client_id = ClientId::new(
+        env::var("GITHUB_CLIENT_ID")
+            .expect("Missing the GITHUB_CLIENT_ID environment variable."),
+    );
+    let github_client_secret = ClientSecret::new(
+        env::var("GITHUB_CLIENT_SECRET")
+            .expect("Missing the GITHUB_CLIENT_SECRET environment variable."),
+    );
+    let auth_url = AuthUrl::new("https://github.com/login/oauth/authorize".to_string())
+        .expect("Invalid authorization endpoint URL");
+    let token_url =
+        TokenUrl::new("https://github.com/login/oauth/access_token".to_string())
+            .expect("Invalid token endpoint URL");
+
+    let client = BasicClient::new(
+        github_client_id,
+        Some(github_client_secret),
+        auth_url,
+        Some(token_url),
+    )
+    .set_redirect_uri(
+        RedirectUrl::new("http://localhost:8080".to_string())
+            .expect("Invalid redirect URL"),
+    );
```

Comment:
Actually had an idea about this, we could spin up a small server in the background to deal with redirect requests locally and close it when no requests to login are present. I can work on this.